project/api/views/views.py

```diff
@@ -203,3 +210,85 @@ class ProfileViewSet(viewsets.ModelViewSet):
     queryset = Profile.objects.all()
     serializer_class = ProfileSerializer
     permission_classes = (IsAuthenticatedOrReadOnly, IsUserOrReadOnly)
+
+
+class HistoryPagination(LimitOffsetPagination):
+    """
+    Pagination for history items
+    """
+
+    page_size_query_param = "limit"
+    max_page_size = 1000
+
+
+class StvHistoryViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    ViewSet for SpacetimeVolume History
+    """
+
+    queryset = HistoricalSpacetimeVolume.objects.all()
+    pagination_class = HistoryPagination
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action == "list":
+            stv = self.request.query_params.get("stv", None)
```

Comment:
Will this return all events, even the _creation_ event? Should we filter it out?

Reply:
It is a subclass of ReadOnlyViewSet, so no creation method is implemented

Reply:
I mean, what should we do with initial geometry upload event? Should it be accessible and shown to user?

Reply:
I think so, in case we want to rollback to the instance's initial state

Reply:
The changes are logged in three categories. Creation, change, deletion. As Liam said creation is used for rolling back to initial version so it is needed I think.